layouts/Base.tsx

```diff
@@ -1,15 +1,73 @@
 'use client';
 
+import classNames from 'classnames';
+import { memo, useMemo, useState } from 'react';
 import type { FC, PropsWithChildren } from 'react';
 
+import WithNavBar from '@/components/withNavBar';
+import WithProgressionSidebar from '@/components/withProgressionSidebar';
+import WithSidebar from '@/components/withSidebar';
+import { usePathname } from '@/navigation.mjs';
 import { NotificationProvider } from '@/providers/notificationProvider';
+import { ShowSidebarProvider } from '@/providers/showSidebarProvider';
 
 import styles from './layouts.module.css';
 
-const BaseLayout: FC<PropsWithChildren> = ({ children }) => (
-  <NotificationProvider viewportClassName="absolute bottom-0 right-0 list-none">
-    <div className={styles.baseLayout}>{children}</div>
-  </NotificationProvider>
-);
+const getSidebar = (slug: string): FC | false => {
+  switch (slug) {
+    case 'learn':
+      return memo(function Sidebar() {
+        return <WithProgressionSidebar navKey="learn" />;
+      });
+    case 'about':
+      return memo(function Sidebar() {
+        return <WithSidebar navKeys={['about', 'getInvolved']} />;
+      });
+    case 'download':
+      return false;
```

Comment:
`false` is not a valid React Component.

Reply:
NULL look better for mailability

Reply:
*maintainability the automatic corrector of safari had done something strange